web/modules/custom/ingredient_term/ingredient_term.module

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @file
+ * Contains hook for Ingredient Term module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function ingredient_term_entity_presave(EntityInterface $entity): void {
+  if ($entity->getEntityTypeId() === 'node' && $entity->getType() === 'article') {
+    $material = $entity->get('field_ingredients')->value;
+    $ingredients = explode(' ', $material);
```

Comment:
The text entered into the textarea can span multiple lines (see the description below the Ingredients field). This will not differentiate between numbers, units, ingredients. Luckily the input strings are validated so they follow a pattern.